common/src/app/common/geom/shapes/modifiers.cljc

```diff
@@ -7,284 +7,279 @@
 (ns app.common.geom.shapes.modifiers
   (:require
    [app.common.data :as d]
-   [app.common.geom.matrix :as gmt]
+   [app.common.data.macros :as dm]
    [app.common.geom.point :as gpt]
-   [app.common.geom.shapes.common :as gco]
    [app.common.geom.shapes.constraints :as gct]
-   [app.common.geom.shapes.layout :as gcl]
-   [app.common.geom.shapes.rect :as gpr]
+   [app.common.geom.shapes.flex-layout :as gcl]
+   [app.common.geom.shapes.pixel-precision :as gpp]
    [app.common.geom.shapes.transforms :as gtr]
-   [app.common.math :as mth]
+   [app.common.pages.helpers :as cph]
+   [app.common.types.modifiers :as ctm]
+   [app.common.types.shape.layout :as ctl]
    [app.common.uuid :as uuid]))
 
-(defn set-pixel-precision
-  "Adjust modifiers so they adjust to the pixel grid"
-  [modifiers shape]
+;;#?(:cljs
+;;   (defn modif->js
+;;     [modif-tree objects]
+;;     (clj->js (into {}
+;;                    (map (fn [[k v]]
+;;                           [(get-in objects [k :name]) v]))
+;;                    modif-tree))))
 
-  (if (and (some? (:resize-transform modifiers))
-           (not (gmt/unit? (:resize-transform modifiers))))
-    ;; If we're working with a rotation we don't handle pixel precision because
-    ;; the transformation won't have the precision anyway
-    modifiers
-
-    (let [center (gco/center-shape shape)
-          base-bounds (-> (:points shape) (gpr/points->rect))
-
-          raw-bounds
-          (-> (gtr/transform-bounds (:points shape) center modifiers)
-              (gpr/points->rect))
-
-          flip-x? (neg? (get-in modifiers [:resize-vector :x]))
-          flip-y? (or (neg? (get-in modifiers [:resize-vector :y]))
-                      (neg? (get-in modifiers [:resize-vector-2 :y])))
-
-          path? (= :path (:type shape))
-          vertical-line? (and path? (<= (:width raw-bounds) 0.01))
-          horizontal-line? (and path? (<= (:height raw-bounds) 0.01))
-
-          target-width (if vertical-line?
-                         (:width raw-bounds)
-                         (max 1 (mth/round (:width raw-bounds))))
-
-          target-height (if horizontal-line?
-                          (:height raw-bounds)
-                          (max 1 (mth/round (:height raw-bounds))))
+(defn resolve-tree-sequence
+  "Given the ids that have changed search for layout roots to recalculate"
+  [ids objects]
 
-          target-p (cond-> (gpt/round (gpt/point raw-bounds))
-                     flip-x?
-                     (update :x + target-width)
+  (assert (or (nil? ids) (set? ids)) (dm/str "tree sequence from not set: " ids))
```

Comment:
Prefer using `us/assert!` macro for documented asserts:
```clojure
(us/assert!
 :expr (or (nil? ids) (set? ids))
 :hint (dm/str "tree sequence from not set: " ids))
 ```

Reply:
ok